updater/spec/dependabot/updater/dependency_group_change_batch_spec.rb

```diff
@@ -54,5 +54,104 @@
+        [
+          Dependabot::DependencyFile.new(
+            name: "Gemfile", content: "mock-updated-gemfile", directory: "/"
+          ),
+          Dependabot::DependencyFile.new(
+            name: "Gemfile.lock", content: "mock-updated-gemfile-lock", directory: "/hello/.."
+          ),
+          Dependabot::DependencyFile.new(
+            name: "Gemfile", content: "mock-updated-package-json", directory: "/elsewhere"
+          ),
+          Dependabot::DependencyFile.new(
+            name: "Gemfile", content: "mock-package-json", directory: "unknown"
+          ),
+          Dependabot::DependencyFile.new(
+            name: "Gemfile", content: "mock-package-json", directory: "../../oob"
+          )
+        ]
+      end
+
+      let(:updated_dependencies) do
+        [
+          Dependabot::Dependency.new(
+            name: "test-dependency-1",
+            package_manager: "bundler",
+            version: "1.1.0",
+            requirements: [
+              {
+                file: "Gemfile",
+                requirement: "~> 1.1.0",
+                groups: ["test"],
+                source: nil
+              }
+            ],
+            directory: "/"
+          ),
+          Dependabot::Dependency.new(
+            name: "test-dependency-2",
+            package_manager: "bundler",
+            version: "1.1.0",
+            requirements: [
+              {
+                file: "Gemfile",
+                requirement: "~> 1.1.0",
+                groups: ["test"],
+                source: nil
+              }
+            ],
+            directory: "/hello/.."
+          )
+        ]
+      end
+
+      let(:group) do
+        Dependabot::DependencyGroup.new(
+          name: "test",
+          rules: { patterns: ["test-*"] }
+        )
+      end
+
+      it "still tracks the updated_dependencies in the global list" do
+        batch = described_class.new(initial_dependency_files: files)
+        batch.merge(dependency_change)
+
+        expect(batch.updated_dependencies).to eq(updated_dependencies)
+      end
+
+      it "tracks the updated dependencies in the appropriate updated_dependency_files" do
+        batch = described_class.new(initial_dependency_files: files)
+        batch.merge(dependency_change)
+
+        # FIXME: All of the files in the dependency_file_batch will contain a copy of the updated dependencies.
+        # Eventually this should be narrowed down so that only the files that were modified
+        # by the newly merged dependency_change have their updated_dependencies updated.
+        dependency_file_batch = batch.instance_variable_get(:@dependency_file_batch)
+        expect(dependency_file_batch["/Gemfile"][:updated_dependencies]).to eq(updated_dependencies)
+        expect(dependency_file_batch["/Gemfile.lock"][:updated_dependencies]).to eq(updated_dependencies)
+        expect(dependency_file_batch["/unknown/Gemfile"][:updated_dependencies]).to eq(updated_dependencies)
```

Comment:
I appreciate the note here as I found the behavior unexpected. I _do_ think it would make sense to expose a method if we need to be able to test this 🙂

Reply:
That's fair! I originally wanted to wait on that for the follow-up PR since the implementation of that method will probably change

I'll do it here though so it's clearer how it will be used in the future

Reply:
Actually, I think I will hold off on making this change!

It will be a modification of the `current_dependency_files` method that gets called GroupUpdateCreation: 

https://github.com/dependabot/dependabot-core/blob/a9ceb8bf5d7daa8bde172404ca23c246591dd4bd/updater/lib/dependabot/updater/group_update_creation.rb#L64

The method and caller will need to be updated to return/accept the dependency files and read the list of updated_dependencies from them. It can then use that list to determine what the `updated_dependency` is without having to scan through the global list of updated_dependencies like it currently does in `#deduce_updated_dependency`: 

https://github.com/dependabot/dependabot-core/blob/a9ceb8bf5d7daa8bde172404ca23c246591dd4bd/updater/lib/dependabot/updater/group_update_creation.rb#L447